scripts/_helpers.py

```diff
@@ -821,3 +829,534 @@ def get_last_commit_message(path):
+    os.chdir(directory_path)
+
+
+def get_current_directory_path():
+    """
+    It returns the current directory path.
+    """
+    return pathlib.Path.cwd()
+
+
+def is_directory_path(path):
+    """
+    It returns True if the path points to a directory.
+
+    False otherwise.
+    """
+    return pathlib.Path(path).is_dir()
+
+
+def is_file_path(path):
+    """
+    It returns True if the path points to a file.
+
+    False otherwise.
+    """
+    return pathlib.Path(path).is_file()
+
+
+def get_relative_path(path, start_path="."):
+    """
+    It returns a relative path to path from start_path.
+
+    Default for start_path is the current directory
+    """
+    return pathlib.Path(path).relative_to(start_path)
+
+
+def path_exists(path):
+    """
+    It returns True if the path exists.
+
+    False otherwise.
+    """
+    return pathlib.Path(path).exists()
+
+
+def create_network_topology(n, prefix, connector=" <-> ", bidirectional=True):
+    """
+    Create a network topology like the power transmission network.
+
+    Parameters
+    ----------
+    n : pypsa.Network
+    prefix : str
+    connector : str
+    bidirectional : bool, default True
+        True: one link for each connection
+        False: one link for each connection and direction (back and forth)
+
+    Returns
+    -------
+    pd.DataFrame with columns bus0, bus1 and length
+    """
+
+    ln_attrs = ["bus0", "bus1", "length"]
+    lk_attrs = ["bus0", "bus1", "length", "underwater_fraction"]
+
+    # TODO: temporary fix for when underwater_fraction is not found
+    if "underwater_fraction" not in n.links.columns:
+        if n.links.empty:
+            n.links["underwater_fraction"] = None
+        else:
+            n.links["underwater_fraction"] = 0.0
+
+    candidates = pd.concat(
+        [n.lines[ln_attrs], n.links.loc[n.links.carrier == "DC", lk_attrs]]
+    ).fillna(0)
+
+    positive_order = candidates.bus0 < candidates.bus1
+    candidates_p = candidates[positive_order]
+    swap_buses = {"bus0": "bus1", "bus1": "bus0"}
+    candidates_n = candidates[~positive_order].rename(columns=swap_buses)
+    candidates = pd.concat([candidates_p, candidates_n])
+
+    def make_index(c):
+        return prefix + c.bus0 + connector + c.bus1
+
+    topo = candidates.groupby(["bus0", "bus1"], as_index=False).mean()
+    topo.index = topo.apply(make_index, axis=1)
+
+    if not bidirectional:
+        topo_reverse = topo.copy()
+        topo_reverse.rename(columns=swap_buses, inplace=True)
+        topo_reverse.index = topo_reverse.apply(make_index, axis=1)
+        topo = pd.concat([topo, topo_reverse])
+
+    return topo
+
+
+def cycling_shift(df, steps=1):
+    """
+    Cyclic shift on index of pd.Series|pd.DataFrame by number of steps.
+    """
+    df = df.copy()
+    new_index = np.roll(df.index, steps)
+    df.values[:] = df.reindex(index=new_index).values
+    return df
+
+
+def download_gadm(country_code, update=False, out_logging=False):
```

Comment:
Why? Is this not already in PyPSA-Earth in build shapes? Why having it twice?

Reply:
Hey @pz-max thanks for noticing it. Actually the two methods are quite different in what they do. I think I need your expertise to understand which one to keep.